### ISSUE TITLE
Only post notifications in slack for main repository, not forks.

### ISSUE THREAD

**Issue by SanketDG** [travis-ci/travis-ci#4722]
Currently, all the fork builds of a repository are also alerted in slack. I am guessing this happens when you are a owner of a organization which has the repository and you also have a fork, both of which use Travis CI.

Is there any way to avert this and only show notifications for the main repository?

**Comment by BanzaiMan** [comment]
So you have an example of this phenomenon? Which fork is sending the notification?

**Comment by SanketDG** [comment]
@BanzaiMan all forks. We have a organization where all the members are "owners" of a repository. So basically, whenever any of us runs their travis build on their own fork, slack posts about the fork build too. I get this is simply because we have the same slack token in all of the `.travis.yml`'s but can we just get notifications whenever our main repository's builds passes or fails?

**Comment by roidrage** [comment]
One solution could be to encrypt the Slack credentials, which is done only for the main repository, so forks can't decrypt the credentials as they don't share the same key. Therefore, they won't send notifications to your Slack channel.

**Comment by SanketDG** [comment]
@roidrage A simple pull from upstream then would again update our forks with the encrypted key.

**Comment by SanketDG** [comment]
@roidrage @BanzaiMan could I get a resolution on this?

**Comment by BanzaiMan** [comment]
Which repository and which build showed this problem? The more recent, the better, so that we can take a look at the logs. Thanks.

**Comment by roidrage** [comment]
@SanketDG Yes, a simple pull from upstream should propagate encrypted credentials to your forks.

**Comment by unitware** [comment]
I can give an example from the open source community.

I have forked https://github.com/betaflight/betaflight into my own github account at https://github.com/unitware/cleanflight (strange naming since betaflight was forked from cleanflight originally, and I used to work in a cleanflight fork originally, not relevant here)

I have travis set up to build my work in progress, but when doing so the travis.yml file from betaflight proper is executed and it contains the notification keys to the betaflightgroup at slack thus the betaflight community is spammed by my work in progress.

When I'm ready I create a pull request to the betaflight repo, now the notifications shall reach the betaflight community.

I do not want to modify the travis.yml since the purpose is to merge back the changes and it would likely give extra work in the merge process (just clicking merge would destroy the notifications in betafligth if I had turned them off in my PR).

One solution would be to let the travis environment variables work also in the travis.yml file like so:
```
notifications:
  slack: ${NOTIFICATION_SLACK}
```

But that did not work when I tested it earlier today.

**Comment by Abhi2424shek** [comment]
Can you guys have a workaround here please? Ideally pinky please? Maybe with an option toogle inbetween. Please?

**Comment by BanzaiMan** [comment]
The easiest way is to ask the upstream to create a new Slack notification string, and encrypt it. The upstream is choosing (perhaps unknowingly) to expose that Slack notification settings.

**Comment by Abhi2424shek** [comment]
Na @BazaiMan that's not helping.

**Comment by Sam-Burns** [comment]
My personal fork of `slimphp/Slim` is spamming `slimphp.slack.com`, which is a nuisance.  Has anyone found a workaround which works?

**Comment by aertmann** [comment]
What BanzaiMan suggest works perfectly fine and should be the way to solve it, been doing that for ages with multiple projects. Would suggest closing the issue.

**Comment by desilinguist** [comment]
I just ran into the same issue for my EducationalTestingService/skll repository. Someone forked it and then I started getting notifications on my Slack channel since my Slack token was in `.travis.yml`. How exactly do I go about doing the encryption thing? It's not entirely clear to me how to encrypt the token and then also use it in `.travis.yml`. Thanks!

**Comment by desilinguist** [comment]
Never mind. I see it: https://docs.travis-ci.com/user/notifications/#Configuring-slack-notifications
